### PR TITLE
feat(logs)!: simplify metadata configuration

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
##### Description

Move the logs metadata configuration to a template. We don't do much more here, there are more changes expected later, pulling out some of the configuration to top-level properties. I did make the configuration for containers and systemd conditional, same for the pipelines for otel log collector and Fluent Bit.

I've added two different ways of customizing the configuration directly. The contents of metadata.metrics.config.merge will be merged into the templated configuration, while metadata.metrics.config.override will replace it in total.

Migration can be found here: https://github.com/SumoLogic/sumologic-kubernetes-tools/pull/387.

---

##### Checklist

- [x] Changelog updated

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -4155,461 +4155,23 @@ metadata:
...
+      ## will change the batch size of the pipeline.
+      ##
+      ## WARNING: This field is not subject to backwards-compatibility guarantees offered by the rest
+      ## of this chart. It involves implementation details that may change even in minor versions.
+      ## Use with caution, and consider opening an issue, so your customization can be added in a safer way.
+      merge: {}
+      ## Completely override existing config and replace it with the contents of this value.
+      ## The value of this key should be a dictionary, that will replace the normal configuration.
+      ## This is an advanced feature, use with caution, and review the generated configuration first.
+      override: {}
```
Comment (kkujawa-sumo) [inline deploy/helm/sumologic/values.yaml:4174]: Please add information about new parameters and remove information about `otelcol.config` in https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/README.md

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/values.yaml:4174]: here: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/docs/fluentd-otc-comparison.md also information about metadata.logs.config should be changed

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/values.yaml:4174]: here: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/docs/best-practices.md#collecting-logs-from-varlogpods I also found the information about `metadata.logs.config`

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/values.yaml:4174]: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/vagrant/values.yaml#L180-L182 should be also adjusted

Comment (kkujawa-sumo) [reply deploy/helm/sumologic/values.yaml:4174]: I found also this config: https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/7b80b8aad023d633c6833242c1ec4b762e12788b/tests/integration/values/values_helm_default_otelcol_logs.yaml#L19-L21

**Comment by kkujawa-sumo** [comment]
Could you add to the changelog below note or something similar?

"Removed explicit configuration for otelcol under `metadata.logs.config`,  add option to merge configuration `metadata.metrics.config.merge` or overwrite default config `metadata.metrics.config.override`"

**Comment by swiatekm-sumo** [comment]
> Could you add to the changelog below note or something similar?
> 
> "Removed explicit configuration for otelcol under `metadata.logs.config`, add option to merge configuration `metadata.metrics.config.merge` or overwrite default configuration `metadata.metrics.config.override`"

I can start a section for log config changes in the migration doc, but it'd just be the start.

File: deploy/helm/sumologic/README.md [hunk 2]
```diff
@@ -369,7 +369,8 @@ The following table lists the configurable parameters of the Sumo Logic chart an
...
-| `metadata.logs.config`                                                    | Configuration for logs otelcol. See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/Configuration.md.                                                                                                                                                                                                                                      | See [values.yaml]                                                                                                 |
+| `metadata.logs.config.merge`                                              | Configuration for logs metadata otelcol, merged with defaults. See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/Configuration.md.                                                                                                                                                                                                       | {}                                                                                                                |
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/README.md:372]:
```suggestion
| `metadata.logs.config.merge`                                              | Configuration for logs metadata otelcol, merged with defaults. See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/configuration.md.                                                                                                                                                                                                       | {}                                                                                                                |
```

File: deploy/helm/sumologic/README.md [hunk 3]
```diff
@@ -369,7 +369,8 @@ The following table lists the configurable parameters of the Sumo Logic chart an
...
+| `metadata.logs.config.override`                                           | Configuration for logs metadata otelcol, replaces defaults.See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/Configuration.md.                                                                                                                                                                                                           | {}                                                                                                                |
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/README.md:373]:
```suggestion
| `metadata.logs.config.override`                                           | Configuration for logs metadata otelcol, replaces defaults.See also https://github.com/SumoLogic/sumologic-otel-collector/blob/main/docs/configuration.md.                                                                                                                                                                                                           | {}                                                                                                                |
```

**Review by astencel-sumo — APPROVED** [review]
:muscle: :trophy:

File: CHANGELOG.md [hunk 4]
```diff
@@ -25,6 +25,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - feat(logs): Changing the default logs metadata provider to otel [#2621]
 - chore!: remove replacing values in configuration marked by 'replace' suffix [#2615]
 - feat(metrics): service name change and switching the metrics provider default to otelcol [#2627]
+- feat!(logs): simplify metadata configuration [#2626]
```
Comment (astencel-sumo) [inline CHANGELOG.md:28]:
```suggestion
- feat(logs)!: simplify metadata configuration [#2626]
```

**Review by kkujawa-sumo — APPROVED** [review]

**Pull request merged**